### ISSUE TITLE
errors of using your pix2pix model

### ISSUE THREAD

**Issue by xingchenzhang** [zhanglichao/generatedTIR_tracking#3]
Hi,

Thank you for your very nice work!

I am now trying to test with your trained model using my RGB data to generate infrared data. But I got the following error: 

"size mismatch for model.model.3.weight: copying a param with shape torch.Size([128, 1, 4, 4]) from checkpoint, the shape in current model is torch.Size([128, 3, 4, 4])."

do you have any idea why this happen and how to solve this?

Thanks a lot!

**Comment by DereenMec** [comment]
我也遇到了相同的问题，楼主最后是怎么解决的？

**Comment by zhanglichao** [comment]
Hi,

Sorry for the late check as these months busy with a submission.
Hope you have solved it at early stage.

As I understand, the input/output of the channels number setting should be equivalent with that the checkpoint model.

I uploaded the 'opt.txt' when I normally used for generating TIR data. Maybe you can refer to it.

best,
Lichao

**Comment by makslevental** [comment]
@zhanglichao 

the opts file isn't correct. here are the model params generated from your opt

```
['model.model.0.weight',
 'model.model.1.model.1.weight',
 'model.model.1.model.2.weight',
 'model.model.1.model.2.bias',
 'model.model.1.model.2.running_mean',
 'model.model.1.model.2.running_var',
 'model.model.1.model.2.num_batches_tracked',
 'model.model.1.model.3.model.1.weight',
 'model.model.1.model.3.model.2.weight',
 'model.model.1.model.3.model.2.bias',
 'model.model.1.model.3.model.2.running_mean',
 'model.model.1.model.3.model.2.running_var',
 'model.model.1.model.3.model.2.num_batches_tracked',
 'model.model.1.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.2.num_batches_tracked',
 'model.model.1.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.3.model.2.num_batches_tracked',
 'model.model.1.model.3.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.num_batches_tracked',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.num_batches_tracked',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.3.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.num_batches_tracked',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.num_batches_tracked',
 'model.model.1.model.3.model.3.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.num_batches_tracked',
 'model.model.1.model.3.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.3.model.3.model.6.num_batches_tracked',
 'model.model.1.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.3.model.6.num_batches_tracked',
 'model.model.1.model.3.model.5.weight',
 'model.model.1.model.3.model.6.weight',
 'model.model.1.model.3.model.6.bias',
 'model.model.1.model.3.model.6.running_mean',
 'model.model.1.model.3.model.6.running_var',
 'model.model.1.model.3.model.6.num_batches_tracked',
 'model.model.1.model.5.weight',
 'model.model.1.model.6.weight',
 'model.model.1.model.6.bias',
 'model.model.1.model.6.running_mean',
 'model.model.1.model.6.running_var',
 'model.model.1.model.6.num_batches_tracked',
 'model.model.3.weight',
 'model.model.3.bias']
```

here are the model params in your checkpoint

```
['model.model.0.weight',
 'model.model.1.model.1.weight',
 'model.model.1.model.2.weight',
 'model.model.1.model.2.bias',
 'model.model.1.model.2.running_mean',
 'model.model.1.model.2.running_var',
 'model.model.1.model.3.model.1.weight',
 'model.model.1.model.3.model.2.weight',
 'model.model.1.model.3.model.2.bias',
 'model.model.1.model.3.model.2.running_mean',
 'model.model.1.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.2.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.1.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.3.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.3.model.4.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.3.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.3.model.5.weight',
 'model.model.1.model.3.model.3.model.6.weight',
 'model.model.1.model.3.model.3.model.6.bias',
 'model.model.1.model.3.model.3.model.6.running_mean',
 'model.model.1.model.3.model.3.model.6.running_var',
 'model.model.1.model.3.model.5.weight',
 'model.model.1.model.3.model.6.weight',
 'model.model.1.model.3.model.6.bias',
 'model.model.1.model.3.model.6.running_mean',
 'model.model.1.model.3.model.6.running_var',
 'model.model.1.model.5.weight',
 'model.model.1.model.6.weight',
 'model.model.1.model.6.bias',
 'model.model.1.model.6.running_mean',
 'model.model.1.model.6.running_var',
 'model.model.3.weight',
 'model.model.3.bias']
```
edit:
ah i guess the only difference is `num_batches_tracked` which must be some sort of difference in how batch norm was implemented when you trained

**Comment by zhanglichao** [comment]
This can be fixed by renaming the parameters in the model dictionary.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by zhanglichao** [comment]
hi,

In the line 59 of the GAN_code/models/base_model.py, change 'model_fix' to model0.
